enhancements/single-node/single-node-openshift-day2-workers.md

```diff
@@ -0,0 +1,290 @@
+node's own IP address. This allows the installation process and the resulting
+cluster to conveniently work without the user having to think about and
+configure DNS (of course external access to the cluster requires the user to
+configure DNS, but this can be done after the installation has completed).
+
+The issue with that approach is that it assumes that `*.apps.<cluster>.<base>`
+should always point at the single control-plane node's IP address. This is
+of-course correct when there's just a single node, but once you start adding
+worker nodes to the cluster it starts causing a potential problem - the
+`router-default` deployment created by the Cluster Ingress Operator, which is
+responsible for load balancing ingress traffic, targets the "worker" pool using
+a node selector. As a result, under some circumstances, the deployment's pods
+may find themselves on the newly added worker nodes, as those also belong to
+the worker pool.
+
+This is a problem because ingress traffic has to be directed at the node
+running the `router-default` pods, and since the DNS entries have been
+"naively" pointed at the original control-plane node which may no longer be
+running those pods, ingress traffic can no longer work. This is until DNS is
+adjusted (solving the problem temporarily) or a load-balancer / some virtual IP
+solution is put in place.
+
+We would like to avoid this complication.
+
+### Goals
+
+- Enable expansion of Single Node clusters by ensuring ingress traffic can always
+  be directed at the single control-plane node successfully, even when new workers
+  are added.
+
+### Non-Goals
+
+- Deal with expansion of the single-node control-plane by adding more control-plane nodes
+- Deal with expansion of clusters that have been installed before the implementation
+  of this enhancement (they can be take ncare of with a simple documentation on how it can be
+  enabled with a few manual steps).
+- Deal with "day 1" UPI installation of single-node control plane clusters
+  that have additional non-control plane worker nodes
+- Deal with cloud IPI installations of Single Node OpenShift. They do not suffer
+  suffer from the floating ingress issue as they're using a cloud provided load
+  balancer which is automatically configured (using a LoadBalancer type Service)
+  to follow the nodes currently holding the router-default pods.
+- Adjust Assisted Installer to use the baremetal platform for single-node installations.
+  The baremetal platforms solves this issue with virtual IP addresses/keepalived.
+- Deal with users who want their ingress traffic to not go through the single
+  control-plane node.
+
+## Proposal
+
+Modify the installer's behavior during Single Node bootstrap-in-place
+installation to create an `IngressController` CR installer manifest targetting
+control-plane nodes, which will prevent the OpenShift Ingress Operator from creating
+the default `IngressController` CR which targets worker nodes. This will ensure
```

Comment:
Perhaps the Ingress Operator should look at the controlPlaneTopolgy and create the `IngressController` CR with the right label selector in case of SingleReplica?

Reply:
This is addressed in the alternatives section, looking at the topology is insufficient